app/Models/Solo/Score.php

```diff
@@ -127,6 +131,32 @@ public function getMode(): string
         return Beatmap::modeStr($this->ruleset_id);
     }
 
+    /**
+     * Queue the item for score processing
+     *
+     * @param array $dataJson Pre-generated json of the score so the transformer doesn't need to be redundantly run
+     * @param bool $force By default only newly created scores are queued. Pass true to force queuing the score
+     */
+    public function queueForProcessing(?array $dataJson = null, bool $force = false): void
+    {
+        if (!$force && !$this->wasRecentlyCreated) {
+            return;
+        }
+
+        $dataJson ??= json_item($this, new ScoreTransformer(ScoreTransformer::TYPE_SOLO));
+
+        LaravelRedis::lpush(static::PROCESSING_QUEUE, json_encode([
+            'ProcessHistory' => null,
+            'Score' => [
+                'beatmap_id' => $dataJson['beatmap_id'],
+                'data' => json_encode($dataJson),
+                'id' => $dataJson['id'],
+                'user_id' => $dataJson['user_id'],
+            ],
+            'TotalRetries' => 0,
```

Comment:
Not that it really matters, but I think this (and `ProcessHistory`) can be fully omitted.